src/cron/arweave/mod.rs

```diff
@@ -1,2 +1,254 @@
+    pub blocks: usize,
+    pub peers: usize,
+    pub queue_length: usize,
+    pub node_state_latency: usize,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct Tag {
+    pub name: String,
+    pub value: String,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct Owner {
+    address: String,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct Fee {
+    winston: String,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct TransactionData {
+    size: String,
+    r#type: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct BlockInfo {
+    pub id: String,
+    pub timestamp: i64,
+    pub height: u128,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct Transaction {
+    pub id: String,
+    pub owner: Owner,
+    pub signature: String,
+    pub recipient: Option<String>,
+    pub tags: Vec<Tag>,
+    pub block: Option<BlockInfo>,
+    pub fee: Fee,
+    pub quantity: Fee,
+    pub data: TransactionData,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct GraphqlNodes {
+    pub node: Transaction,
+    pub cursor: String,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GraphqlEdges {
+    pub edges: Vec<GraphqlNodes>,
+    pub page_info: PageInfo,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PageInfo {
+    pub has_next_page: bool,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct TransactionsGqlResponse {
+    pub transactions: GraphqlEdges,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct GraphqlQueryResponse {
+    pub data: TransactionsGqlResponse,
+}
+
+#[derive(Deserialize, Serialize, Default, Clone)]
+pub struct TransactionStatus {
+    pub block_indep_hash: String,
+}
+
+#[derive(Clone)]
+pub enum ArweaveProtocol {
+    Http,
+    Https,
+}
+
+#[derive(Clone)]
+pub struct Arweave {
+    pub host: String,
+    pub port: u16,
+    pub protocol: ArweaveProtocol,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct GqlVariables {
+    pub owners: Vec<String>,
+    pub first: u128,
+    pub after: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ReqBody {
+    pub query: String,
+    pub variables: GqlVariables,
+}
+
+pub trait ArweaveContext {
+    fn get_client(&self) -> reqwest::Client;
+}
+
+#[warn(dead_code)]
+impl Arweave {
+    pub fn new(port: u16, host: String, protocol: String) -> Arweave {
+        Arweave {
+            port,
+            host,
+            protocol: match &protocol[..] {
+                "http" => ArweaveProtocol::Http,
+                "https" | _ => ArweaveProtocol::Https,
+            },
+        }
+    }
+
+    pub async fn get_tx_data<Context>(
+        &self,
+        transaction_id: &str,
+        ctx: &Context,
```

Comment:
Minor thing, but I would move context as the first arg (apart from self). Usually context or environment kind of args are passed before actual input args. Not really a big thing, but just common ordering what you can see also elsewhere.